src/firefly/java/edu/caltech/ipac/util/download/URLDownload.java

```diff
@@ -45,6 +49,30 @@
     public static String getUserFromUrl(String url) { return getUserInfoPart(url,0); }
     public static String getPasswordFromUrl(String url) { return getUserInfoPart(url,1); }
 
+    public static String getUserAgentString() {
+        Version v= VersionUtil.getAppVersion();
+        String fUA= getFireflyUA(v);
+        if (v.getAppName().equalsIgnoreCase("firefly")) return fUA;
+        if (v.getMajor()==0) return fUA;
+        return fUA + " ("+v.getAppName() + "/" + v.getMajor()+"."+v.getMinor()+"."+v.getRev() + ")";
+    }
+
+    private static String getFireflyUA(Version v) {
+        String unknown= "Firefly/development";
+        String tag = v.getBuildGitTagFirefly();
+        if (StringUtils.isEmpty(tag)) return unknown;
+        Matcher m = Pattern.compile("[1-9]").matcher(tag);
+        if (!m.find()) return unknown;
+        int firstIdx= m.start();
+        int lastIdx= m.start();
+        while (m.find()) lastIdx= m.start();
+        try {
+            return "Firefly/" + tag.substring(firstIdx,lastIdx+1);
+        } catch (Exception e) {
```

Comment:
This logic may leads to wrong value.  It ignores the trailing 0s which is significant.  Try something like this:
`jshell> Pattern.compile("\\d+(\\.\\d+)+").matcher("release-2021.13.30.dev").results().findFirst().get().group();`

Reply:
I did forget the `0` but you are so much better at regular expressions. I am sure this is much better but I will play with it to understand.